src/main/java/software/sigma/bu003/internship/vedmid_andrii/service/PartService.java

```diff
@@ -0,0 +1,75 @@
+package software.sigma.bu003.internship.vedmid_andrii.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.sigma.bu003.internship.vedmid_andrii.entity.Part;
+import software.sigma.bu003.internship.vedmid_andrii.exception.PartNotFoundException;
+import software.sigma.bu003.internship.vedmid_andrii.repository.PartRepository;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@AllArgsConstructor
+public class PartService {
+
+    private final PartRepository partRepository;
+
+    public Part createPart(Part part) {
+        checkIsPart(part.getBrand(), part.getCode());
+
+        Part partFromDB = partRepository.findByBrandAndCode(part.getBrand(), part.getCode());
+
+        if(!Objects.isNull(partFromDB)){
+            return partFromDB;
+        } else {
+            return partRepository.save(part);
+        }
+    }
+
+    public List<Part> getAllParts() {
+        return partRepository.findAll();
+    }
+
+    public Part getPart(String brand, String code) {
+        Part partFromDB = partRepository.findByBrandAndCode(brand, code);
```

Comment:
all you need is -         
![image](https://user-images.githubusercontent.com/13138548/201091558-22109eea-b98c-40a1-a4bd-f3c231b56c8f.png)

Reply:
also you need to add brand and code in PartNotFoundException error msg